handout/routing/route_guards.md

```diff
@@ -0,0 +1,115 @@
+# Route Guards #
+
+Route guards provide a way to control whether the user can navigate to or navigate away from a given route.
+
+For example, we may want some routes to only be accessible once the user has logged in or accepted Terms & Conditions. We can use route guards to check these conditions and control access to routes.
+
+Route guards can also control whether a user can leave a certain route. For example, say the user has typed information into a form on the page, but has not submitted the form. If they were to leave the page, they would lose the information. We may want to prompt the user if the user attempts to leave the route without submitting or saving the information.
+
+## Registering the route guards with routes ##
+
+In order to use route guards, we must register them with the specific routes we want them to run for.
+
+For example, say we have an `accounts` route that only users that are logged in can navigate to. This page also has forms and we want to make sure the user has submitted unsaved changes before leaving the accounts page.
+
+In our route config we can add our guards to that route:
+
+```javascript
+import { LoginRouteGuard } from './login-route-guard';
+import { SaveFormsGuard } from './save-forms-guard';
+
+import { provideRouter, RouterConfig } from '@angular/router';
+import { HomePage } from './home-page';
+import { AccountPage } from './account-page';
+
+const routes: RouterConfig = [
+  { path: 'home', component: HomePage },
+  {
+    path: 'accounts',
+    component: AccountPage,
+    canActivate: [LoginRouteGuard],
+    canDeactivate: [SaveFormsGuard]
+  }
+];
+
+bootstrap(AppComponent, [
```

Comment:
```js
bootstrap(AppComponent, [
   provideRouter(routes),
   LoginRouteGuard,
   SaveFormsGuard
]);
```

Probably worth calling out that guards are injectable services, and need to be registered as a provider with the application.